setup.py

```diff
@@ -29,6 +29,33 @@
 # setuptools
 from setuptools import setup, find_packages
 
+# Specify groups of optional dependencies
+extras_require = {
+    'machine_learning': [
+        'gensim',
+        'numpy',
+        'pyparsing',
```

Comment:
If we want to be pedantic then pyparsing should be moved to some other section ("tgrep"?) because neither pyparsing nor tgrep are related to machine learning.